lib/components/text_component.dart

```diff
@@ -30,15 +35,23 @@ class TextComponent extends PositionComponent {
   }
 
   void _updateBox() {
-    final TextPainter tp = config.toTextPainter(text);
-    width = tp.width;
-    height = tp.height;
+    _tp = config.toTextPainter(_text);
+    width = _tp.width;
+    height = _tp.height;
+
+    _updateOffset(Position.empty());
+  }
+
+  void _updateOffset(Position p) {
+    final Position translatedPosition =
+        anchor.translate(p, Position.fromSize(_tp.size));
+    _translatedOffset = translatedPosition.toOffset();
   }
 
   @override
   void render(Canvas c) {
     prepareCanvas(c);
-    config.render(c, text, Position.empty());
+    _tp.paint(c, _translatedOffset);
```

Comment:
`TextConfig.render` is pretty much useless after this.

Reply:
@renancaraujo you can use TextConfig outside TextComponents as well.